pkg/rest/apisurface.go

```diff
@@ -172,8 +172,8 @@ func unpackDeprovisionRequest(r *http.Request) (*osb.DeprovisionRequest, error)
 
 	vars := mux.Vars(r)
 	osbRequest.InstanceID = vars[osb.VarKeyInstanceID]
-	osbRequest.ServiceID = vars[osb.VarKeyServiceID]
-	osbRequest.PlanID = vars[osb.VarKeyPlanID]
+	osbRequest.ServiceID = r.URL.Query().Get(osb.VarKeyServiceID)
```

Comment:
I would prefer this be [`r.FormValue`](https://golang.org/pkg/net/http/#Request.FormValue). IMO this helper method makes it more clear.

Reply:
I agree with you, it's worked well with the handler in our Automation Broker. I didn't know what the best way of doing that was. Do I do it here on this one change? Or would it be better to do it for all cases at once?

Reply:
Alternatively I can fix all of the cases in this same PR.

Reply:
Let’s fix the bug here and do the helper method thing everywhere in another PR